File: Data Structures & Algorithms/4sum/submission-0.py
class Solution:
    def fourSum(self, nums: List[int], target: int) -> List[List[int]]:
        n = len(nums)
        output = set()

        for i in range(n-3):
            for j in range(i+1, n-2):
                for k in range(j+1, n-1):
                    for l in range(k+1, n):
                        if nums[i] + nums[j] + nums[k] + nums[l] == target:
                            temp = [nums[i], nums[j], nums[k], nums[l]]
                            state = tuple(sorted(temp))
                            output.add(state)
        
        return list(output)

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        for(int i=0; i<prices.length-1; i++){
            for(int j=i; j<prices.length; j++){
                int temp = prices[j]-prices[i];
                if(temp > profit){
                    profit = temp;
                }
            }
        }

        return profit;
    }
}
